binary_search_tree/binary_search_tree.py

```diff
@@ -7,18 +7,36 @@ def __init__(self, value):
     self.right = None
 
   def insert(self, value):
-    bst = BinarySearchTree(value)
-    if bst.left and bst.right is None:
-      root = bst
+    if self.left and self.right and self.value is None:
```

Comment:
I'm not sure this is going to run. if `self.left` is True, it's value cannot be `None`